core/types/rlp_payload.libevm.go

```diff
@@ -68,9 +87,12 @@ func TestOnlyClearRegisteredExtras() {
 var registeredExtras register.AtMostOnce[*extraConstructors]
 
 type extraConstructors struct {
-	stateAccountType  string
-	newStateAccount   func() *pseudo.Type
-	cloneStateAccount func(*StateAccountExtra) *StateAccountExtra
+	stateAccountType           string
+	newHeader, newStateAccount func() *pseudo.Type
+	cloneStateAccount          func(*StateAccountExtra) *StateAccountExtra
+	hooks                      interface {
+		hooksFromHeader(*Header) HeaderHooks
+	}
```

Comment:
It can be done by just setting it to `ExtraPayloads[H,SA].hooksFromHeader` if we wanted. Much of this code is replicated directly from the first time I did it in `params` and there the interface has more than one method so it was cleaner + easier to maintain.

Reply:
It could had been nice to have `.hooksFromHeader` directly instead of `hooks.hooksFromHeader` which repeats itself a bit. Why is the interface form easier to maintain versus function fields by the way? 🤔